tools/fix_candidate_age.php

```diff
@@ -0,0 +1,135 @@
+ * the NDB_BVL_Instrument::_saveValues() function to safely make changes to ages.
+ *
+ * Usage
+ *  - php fix_candidate_age.php;
+ *  - php fix_candidate_age.php confirm;
+ *
+ * PHP Version 7
+ *
+ * @category Main
+ * @package  Loris
+ * @author   Various <example@example.com>
+ * @license  Loris license
+ * @link     https://www.github.com/aces/Loris-Trunk/
+ */
+require_once 'generic_includes.php';
+
+$config   = NDB_Config::singleton();
+$database = $config->getSetting('database');
+
+$instruments = Utility::getAllInstruments();
+
+$confirm =false;
+if (!empty($argv[1]) && $argv[1] == 'confirm') {
+    $confirm = true;
+}
+
+$incorrectAges = array();
+$nullAges      = array();
+
+foreach ($instruments as $inst=>$fullName) {
+
+    if (!$DB->tableExists($inst)) {
+        echo $inst." does not have a table in the Database\n";
+        continue;
+    } else if (strpos('_proband',$inst) !== false) {
+        echo $inst." is a Proband instrument and should be handled seperately.\n";
+        continue;
+    }
+
+    //Get instrument SQL table
+    $DBInstTable = $DB->pselect(
+        "SELECT * 
+         FROM $inst i 
+            LEFT JOIN flag f ON (i.commentID=f.CommentID)
+            LEFT JOIN session s ON (f.SessionID=s.ID)
+            LEFT JOIN candidate c ON (s.CandID=c.CandID)
+         WHERE c.Active='Y' AND s.Active='Y'",
+        array()
+    );
+
+    // Break if Date taken does not exist
+    if (!$DB->columnExists($inst, 'Date_taken')) {
+        echo $inst." does not use a `Date_taken` field and should be handled seperately.\n";
+        continue;
+    }
+
+    foreach ($DBInstTable as $k => $row) {
+        // Get Instrument Instance
+        try {
+            $instrument = NDB_BVL_Instrument::factory($inst, $row['CommentID'], null, false);
+        } catch (LorisException $e) {
+            echo "$inst instrument row with CommentID: ".$row['CommentID']." was ".
+                " Ignored for one of the following reasons:\n".
+                "  - The candidate is inactive.\n".
+                "  - The session is inactive.\n\n";
+            continue;
+        }
+        if (!$instrument) {
+            // instrument does not exist
+            continue;
+        }
+
+        $dateTaken = $row['Date_taken'];
+        $commentID = $row['CommentID'];
+
+        // Flag for problem with date
+        $trouble =false;
+        if (!empty($row['Date_taken'])) {
+            // Check if age is null OR if wrong age
+            if (empty($row['Candidate_Age'])) {
+                // Null age
+                $nullAges[$inst][$commentID] = $row;
+                $trouble =true;
+            } else {
+                // get Age from instrument class
+                $calculatedAge       = $instrument->getCandidateAge();
+                $agemonths           = $calculatedAge['year'] * 12 + $calculatedAge['mon'] + ($calculatedAge['day'] / 30);
+                $calculatedAgeMonths = (round($agemonths*10) / 10.0);
```

Comment:
@driusan @ridz1208 Glad to see these formulas are the same as in the instrument library.